JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -61,11 +61,13 @@ public async Task PublishAsync(Message message, CancellationToken cancellationTo
                 var response = await Client.PublishAsync(request, cancellationToken).ConfigureAwait(false);
                 _eventLog.LogInformation($"Published message: '{request.Subject}' with content {request.Message}");
 
-                MessageResponseLogger?.Invoke(new MessageResponse
-                {
-                    HttpStatusCode = response?.HttpStatusCode,
-                    MessageId = response?.MessageId
-                }, message);
+                var responseData = new MessageResponse
```

Comment:
So the old code ` MessageResponseLogger?.Invoke(new MessageResponse...` would only allocate a `new MessageResponse` if it passed the null check? Agreed we should keep that if so.

Reply:
I believe the under the hood code generated by the compiler would have been:

```csharp
if (MessageResponseLogger != null)
{
    MessageResponseLogger(responseData, new() { });
}
```

So yes, it would spare the allocation.